### ISSUE TITLE
websocket maintains params between requests...

### ISSUE THREAD

**Issue by heavyk** [evantahler/actionhero#628]
a simple code example:
```
var logC = document.getElementById('log');
    
function log(msg){
    var c = document.createElement('pre');
    c.innerHTML = msg;
    logC.appendChild(c);
}

var client = new actionheroClient({
	url: 'http://localhost:8080'
});
client.connect(function() {
	log('client connected');
    client.actionWebSocket({action: 'get_feed', category:"5521b5665c6b1c71c8800e4e"}, function(err, res) {
		log('first request:\n' + JSON.stringify(res, null, '\t'));
        client.actionWebSocket({action: 'get_feed'}, function(err, res2) {
            log('second request:\n' + JSON.stringify(res2, null, '\t'));
        });
	});
});
log('client created');
```
(doesn't matter if I use .action, .actionWebSocket or .actionWeb)
both requests will have category defined in the action on the server. I've been trying to figure it out, but have not been able to.  even if I print out connection.params in `handleData` function in servers/websocket.js:221 I can see that the connection.params.category has persisted.

am I missing something?

**Comment by evantahler** [comment]
You should be using `client.action(actionName, data)`.



--
 *Evan Tahler* | evantahler@gmail.com | +1 (415) 413-8268
 evantahler.com | actionherojs.com

**Comment by heavyk** [comment]
I just started AH on port 8880, copied the sleeptest.js action, and changed the code per your suggestion to `client.action(...)`

a fully working example

a fiddle: https://jsfiddle.net/0bckekto/4/

the code:
```
var logC = document.getElementById('log');
    
function log(msg){
    var c = document.createElement('pre');
    c.innerHTML = msg;
    logC.appendChild(c);
}

var client = new ActionheroClient({
	url: 'http://localhost:8880'
});
client.connect(function() {
	log('client connected');
    client.action('sleepTest', {sleepDuration: 2000}, function(res) {
		log('first request:\n' + JSON.stringify(res, null, '\t'));
        client.action('sleepTest', {sleepDuration: void(0)}, function(res2) {
            log('second request:\n' + JSON.stringify(res2, null, '\t'));
            client.action('sleepTest', function(res3) {
                log('third request:\n' + JSON.stringify(res3, null, '\t'));
            });
        });
	});
});
log('client created');
```

the result:
```
client created

client connected

first request:
{
	"sleepStarted": 1428359221480,
	"sleepEnded": 1428359223480,
	"sleepDelta": 2000,
	"sleepDuration": 2000,
	"messageCount": 2,
	"context": "response"
}

second request:
{
	"sleepStarted": 1428359223482,
	"sleepEnded": 1428359225482,
	"sleepDelta": 2000,
	"sleepDuration": 2000,
	"messageCount": 3,
	"context": "response"
}

third request:
{
	"sleepStarted": 1428359225484,
	"sleepEnded": 1428359227484,
	"sleepDelta": 2000,
	"sleepDuration": 2000,
	"messageCount": 4,
	"context": "response"
}
```

as you can see, all requests are 2000ms. if I change the duration in the second request to 500, then the third will be 500ms as well.

**Comment by evantahler** [comment]
I cannot reproduce the behavior you are seeing.  In a *fresh* actionhero install, try your commands again.  You can see that both serial and parallel work as expected:

**parallel**

```javascript
var client = new ActionheroClient({
	url: 'http://localhost:8080'
});

client.connect(function(){
	client.action('sleepTest', {sleepDuration: 3000}, function(data){ 
		console.log('response a: ' + data.sleepDuration); 
	});
	client.action('sleepTest', {sleepDuration: 2000}, function(data){ 
		console.log('response b: ' + data.sleepDuration); 
	});
	client.action('sleepTest', {sleepDuration: 1000}, function(data){ 
		console.log('response c: ' + data.sleepDuration); 
	});
});

// RESULT //
response c: 1000
response b: 2000
response a: 3000
```

**series**

```javascript
var client = new ActionheroClient({
	url: 'http://localhost:8080'
});

client.connect(function(){
	client.action('sleepTest', {sleepDuration: 3000}, function(data){ 
		console.log('response a: ' + data.sleepDuration);  
		client.action('sleepTest', {sleepDuration: 2000}, function(data){ 
			console.log('response b: ' + data.sleepDuration);  
			client.action('sleepTest', {sleepDuration: 1000}, function(data){ 
				console.log('response c: ' + data.sleepDuration);  
			});
		});
	});
});


// RESULT //

response a: 3000
response b: 2000
response c: 1000
```

You can also try this in the console on demo.actionherojs.com, which always has the latest master branch deployed to it.

**Comment by evantahler** [comment]
Ahh! the void/undefined is the problem.  Reopening...

**Comment by evantahler** [comment]
Oh.  so **by design** (as clarified in [this test](https://github.com/evantahler/actionhero/commit/d4296177ee03101a52c56805144ee53635709b84)) we **wanted** WS params to be 'sticky' at some point.  This means that if you don't overwrite a param explicitly, the old value will be "stuck" to your connection.  

The idea was that you only ever needed to specify `{userId: 1}` in one action, and it would be there going forward.  I agree with you, and I think that this is a bad idea.  [I'll remove it in v11](https://github.com/evantahler/actionhero/pull/629).  

Sticky params do however, still make sense for the `socket` server API, which uses the `addParam` api

**Comment by heavyk** [comment]
I figured it was something along that line, great... actually that could potentially be very useful to not neet to send the token every time. so, perhaps there could be a way to instead define which params are sticky?

so, we definitely need to be able to undefine some parameters in some way... is there a quick fix we can introduce? actually, I just realized it may be as easy as looking on the rawConnection and undefine everything that's not the token for the duration of this version. I'll try that now. ok cool :+1:
